site/functions/server/main.go

```diff
@@ -0,0 +1,272 @@
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		writeJSON(w, PluginsResponse{Error: ErrorResponse{Message: fmt.Sprintf("error retrieving repo contents: %v", err)}})
+		return
+	}
+	log.Printf("github response=%s rate: limit=%d remaining=%d",
+		resp.Status, resp.Rate.Limit, resp.Rate.Remaining)
+	var out PluginsResponse
+
+	plugins, err := fetchPlugins(ctx, filterYAMLs(dir))
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		writeJSON(w, PluginsResponse{Error: ErrorResponse{Message: fmt.Sprintf("failed to fetch plugins: %v", err)}})
+		return
+	}
+
+	for _, v := range plugins {
+		pi := pluginInfo{
+			Name:             v.Name,
+			Homepage:         v.Spec.Homepage,
+			ShortDescription: v.Spec.ShortDescription,
+			GithubRepo:       findRepo(v.Spec.Homepage),
+		}
+		out.Data.Plugins = append(out.Data.Plugins, pi)
+	}
+
+	w.Header().Set("Cache-Control", fmt.Sprintf("public, max-age=%d", cacheSeconds))
+	writeJSON(w, out)
+}
+
+func fetchPlugins(ctx context.Context, entries []*github.RepositoryContent) ([]*krew.Plugin, error) {
+	var (
+		mu  sync.Mutex
+		out []*krew.Plugin
+	)
+
+	queue := make(chan string)
+	g, ctx := errgroup.WithContext(ctx)
+
+	for i := 0; i < pluginFetchWorkers; i++ {
+		g.Go(func() error {
+			for url := range queue {
+				p, err := readPlugin(url)
+				if err != nil {
+					return err
+				}
+				mu.Lock()
+				out = append(out, p)
+				mu.Unlock()
+			}
+			return nil
+		})
+	}
+
+	for _, v := range entries {
+		url := v.GetDownloadURL()
+		select {
+		case <-ctx.Done():
+			break
+		case queue <- url:
+		}
+	}
+
+	close(queue)
+
+	return out, g.Wait()
+}
+
+func readPlugin(url string) (*krew.Plugin, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get %s", url)
+	}
+	defer resp.Body.Close()
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to read file %s: %w", url)
+	}
+
+	var v krew.Plugin
+	if err = yaml.Unmarshal(b, &v); err != nil {
+		return nil, errors.Wrapf(err, "failed to parse plugin manifest for %s", url)
+	}
+	return &v, nil
+}
+
+func findRepo(homePage string) string {
+	if matches := githubRepoPattern.FindStringSubmatch(homePage); matches != nil {
+		return matches[1]
+	}
+
+	knownHomePages := map[string]string{
+		`https://krew.sigs.k8s.io/`:                                  "kubernetes-sigs/krew",
+		`https://sigs.k8s.io/krew`:                                   "kubernetes-sigs/krew",
+		`https://kubernetes.github.io/ingress-nginx/kubectl-plugin/`: "kubernetes/ingress-nginx",
+		`https://kudo.dev/`:                                          "kudobuilder/kudo",
+		`https://kubevirt.io`:                                        "kubevirt/kubectl-virt-plugin",
+		`https://popeyecli.io`:                                       "derailed/popeye",
+		`https://soluble-ai.github.io/kubetap/`:                      "soluble-ai/kubetap",
+	}
```

Comment:
@corneliusweig I wonder if we should move this to somewhere like a `var` declaration at the top. I suspect it'll get updated frequently. :D

Reply:
I wonder if we should even go one step further and extract this configuration into non-code. In `krew-index-tracker`, I also need this list and it is error-prone to redeploy every time this changes.

Maybe we can keep this map in a very dumb format in a text file inside `krew-index` or `krew`. Something like
```txt
https://krew.sigs.k8s.io/ kubernetes-sigs/krew
https://sigs.k8s.io/krew kubernetes-sigs/krew
...
```

Reply:
I'm -1 for a new file/format.
I think we can hardcode these still in a centralized source code file.
We can do a variable on top of main.go for now.